js/recipe/introduction.js

```diff
@@ -0,0 +1,17 @@
+import { recipeParentElement } from "../constants.js"
+import { render } from "../htmlRenderer.js"
+import * as els from "../htmlRenderer.js"
+
+export function introduction(recipe) {
+  const firstParagraphElement = els.p({innerHTML: recipe.description[0]});
+  const firstParagraphDiv = els.div({className: "first-paragraph", children: [firstParagraphElement]});
+  const latterDescriptionParagraphs = recipe.description.slice(1);
+
+  const latterParagraphElements = latterDescriptionParagraphs.map(paragraph => {
+    return els.p({innerHTML: paragraph});
+  });
+
+  const children = [firstParagraphDiv].concat(latterParagraphElements);
+
+  render(recipeParentElement, els.div({className: "introduction-paragraphs", children: children}));
```

Comment:
Might be interesting to see what this looks like without the intermediate variables:

```js
els.div({ className: "introduction-paragraphs", children: [
    els.div({ className: "first-paragragh", children: [
      els.p({ innerHTML: recipe.description[0] })
      ]}),
    ...latterDescriptionParagraphs.map(paragraph => els.p({ innerHTML: paragraph })
    )
  ]});
```

(taking some liberties with indentation) and compare it to the JSX version:

```jsx
<div className="introduction-paragraphs">
  <div className="first-paragraph">
    <p>{recipe.description[0]}</p>
  </div>
  {latterDescriptionParagraphs.map(paragraph => (<p>{paragraph}</p>))
</div>
```

This also suggests to me that always needing `children` to be a list is annoying and could be changed (we could check inside `render` if `children` is a list and wrap it if not).

Reply:
And it might be useful to think about how the top thing would be implemented in Ruby where we have blocks:

```ruby
div(className: "introduction-paragraphs") do
  div(className: "first-paragraph") do
    p { recipe.description[0] }
  end
  ...
end
```

We've actually implemented a DSL in JS, which is sometimes more natural in Ruby because it has blocks (which allow us to express parent-child relationships like HTML elements).

Reply:
That's really cool that you can see the similarities with JSX! I suppose it's not too surprising since you have been nudging us along gradually to a React-like framework :)

Reply:
When you say it's annoying that children has to be a list and we could put a check in `render`, do you mean that we could pass an array of arrays to `render` and it would wrap them into a single array?

Reply:
Also - is "list" the correct term here rather than "array" ?

Reply:
Ah yeah I'm just being loose with array/list, it's an [Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array) in JS.
re: children in a list, I was thinking that it's quite annoying to always wrap them in a list, even if there's one child e.g.:

```js
els.div({ className: "foo", children: [els.div({ className: "bar" })] })
```

We could allow `render` to accept `children` as either being an element or an array of elements. That would mean `render` has to check the type of `children` and handle each case slightly differently.

Then again, it's no big deal to wrap things in a list really :)

Reply:
I'm going to make a note to change this behaviour of the children in htmlRenderer.js in a future PR, and get this one merged.